mono/metadata/marshal.c

```diff
@@ -10013,6 +10017,76 @@ get_virtual_stelemref_wrapper (int kind)
 		mono_mb_emit_exception (mb, "ArrayTypeMismatchException", NULL);
 		break;
 	}
+
+	case STELEMREF_CLASS_SMALL_IDEPTH:
+		/*
```

Comment:
I assume this works because the array has a minimum size so we can't index out of it ?